tests/integration_tests/modules/test_set_hostname.py

```diff
@@ -30,35 +30,100 @@
 fqdn: cloudinit2.test.io
 """
 
+REQUIRES_FILE_FLAG = PLATFORM == "gce" and CURRENT_RELEASE >= NOBLE
+
 
 @pytest.mark.ci
 class TestHostname:
     @pytest.mark.user_data(USER_DATA_HOSTNAME)
     def test_hostname(self, client):
         hostname_output = client.execute("hostname")
-        assert "cloudinit2" in hostname_output.strip()
+        if REQUIRES_FILE_FLAG:
+            assert "cloudinit2" not in hostname_output.strip()
+        else:
+            assert "cloudinit2" in hostname_output.strip(), CURRENT_RELEASE
 
     @pytest.mark.user_data(USER_DATA_PREFER_FQDN.format(True))
     def test_prefer_fqdn(self, client):
         hostname_output = client.execute("hostname")
-        assert "cloudinit2.test.io" in hostname_output.strip()
+        if REQUIRES_FILE_FLAG:
+            assert "cloudinit2.test.io" not in hostname_output.strip()
+        else:
+            assert "cloudinit2.test.io" in hostname_output.strip()
 
     @pytest.mark.user_data(USER_DATA_PREFER_FQDN.format(False))
     def test_prefer_short_hostname(self, client):
         hostname_output = client.execute("hostname")
-        assert "cloudinit1" in hostname_output.strip()
+        if REQUIRES_FILE_FLAG:
+            assert "cloudinit1" not in hostname_output.strip()
+        else:
+            assert "cloudinit1" in hostname_output.strip()
 
     @pytest.mark.user_data(USER_DATA_FQDN)
     def test_hostname_and_fqdn(self, client):
+        hostname_output = client.execute("hostname")
+        fqdn_output = client.execute("hostname --fqdn")
+        host_output = client.execute("grep ^127 /etc/hosts")
+
+        assert "127.0.0.1 localhost" in host_output
+        if REQUIRES_FILE_FLAG:
+            assert "cloudinit1" not in hostname_output.strip()
+            assert "cloudinit2.i9n.cloud-init.io" not in fqdn_output.strip()
+            assert (
+                f"127.0.1.1 {fqdn_output} {hostname_output}" not in host_output
+            )
+        else:
+            assert "cloudinit1" in hostname_output.strip()
+            assert "cloudinit2.i9n.cloud-init.io" in fqdn_output.strip()
+            assert f"127.0.1.1 {fqdn_output} {hostname_output}" in host_output
+
+
+USER_DATA_HOSTNAME_FILE = """\
+#cloud-config
+hostname: cloudinit2
+create_hostname_file: true
+"""
+
+USER_DATA_FQDN_FILE = """\
+#cloud-config
+manage_etc_hosts: true
+hostname: cloudinit1
+fqdn: cloudinit2.i9n.cloud-init.io
+create_hostname_file: true
+"""
+
+USER_DATA_PREFER_FQDN_FILE = """\
+#cloud-config
+prefer_fqdn_over_hostname: {}
+hostname: cloudinit1
+fqdn: cloudinit2.test.io
+create_hostname_file: true
+"""
+
+
+class TestCreateHostnameFile:
```

Comment:
Do you think we need this test when `False == REQUIRES_FILE_FLAG`? iirc this key is a no-op there, right?

Reply:
It's testing both here. I think it's good to still check that the file is actually created and applied successfully.

...or am I misunderstanding your question?